src/main/java/dev/blaauwendraad/masker/json/config/KeyMaskingConfig.java

```diff
@@ -0,0 +1,377 @@
+package dev.blaauwendraad.masker.json.config;
+
+import javax.annotation.CheckForNull;
+import java.nio.charset.StandardCharsets;
+import java.util.Objects;
+
+public class KeyMaskingConfig {
+    @CheckForNull
+    private final byte[] maskStringsWith;
+    @CheckForNull
+    private final byte[] maskStringCharactersWith;
+    private final boolean disableNumberMasking;
+    @CheckForNull
+    private final byte[] maskNumbersWith;
+    @CheckForNull
+    private final byte[] maskNumberDigitsWith;
+    private final boolean disableBooleanMasking;
+    @CheckForNull
+    private final byte[] maskBooleansWith;
+
+    KeyMaskingConfig(
```

Comment:
Should it also accept an `jsonPathKey` option on a key configuration level? This way we can solve the bug you mentioned yesterday by the solution you implemented of having a key masking config in a TrieNode.
Also, this way we won't have to make assumptions of wether the key is json path or not based on wether it starts with "$." or "$[" sequences.

Reply:
I've been thinking about it a bit and I think it's not that important to solve that bug as it's pretty rare that it could happen. In order to fix it I would need to also split configs by whether that's a key or a json paths (will complicate a code a bit), or, as we discussed, associate config with a specific key (would make our fluent API slightly worse)